crates/sncast/src/main.rs

```diff
@@ -138,6 +138,9 @@ fn main() -> Result<()> {
         )
         .expect("Failed to build script");
         let metadata_with_deps = get_scarb_metadata_with_deps(&manifest_path)?;
+
+        runtime.block_on(warn_if_incompatible_rpc_version(&provider, &config.rpc_url))?;
```

Comment:
let's move this higher up, right after config & provider, becase if there are incompatible versions they may result in errors, so we may never even see this warning.